x-pack/plugins/maps/public/embeddable/map_embeddable.tsx

```diff
@@ -152,6 +153,23 @@ export class MapEmbeddable
       this.onFatalError(e);
       return;
     }
+    try {
```

Comment:
Instead of handing saved object check seperatly in MapEmbeddable and MapApp, how about moving logic into a new SavedMap called something like "hasSavedObjectIdConflict"? That way all checking logic can just be part of SavedMap and consolidated into a single location.